histolab/tiler.py

```diff
@@ -1121,3 +1121,292 @@ def _scores(
+            filenames.append(tile_filename)
+            logger.info(
+                f"\t Tile {tiles_counter} - score: {score} saved: {tile_filename}"
+            )
+
+        if report_path:
+            self._save_report(
+                report_path,
+                filtered_tiles_scores,
+                filtered_tiles_scaled_scores,
+                filenames,
+            )
+
+        if tiles_counter > 0:
+            tiles_counter += 1
+
+        logger.info(f"{tiles_counter} Grid Tiles have been saved.")
+
+    # ------- implementation helpers -------
+
+    def _tiles_generator(
+        self, slide: Slide, extraction_mask: BinaryMask = BiggestTissueBoxMask()
+    ) -> Tuple[List[Tuple[float, CoordinatePair]], List[Tuple[float, CoordinatePair]]]:
+        r"""Calculate the tiles with scores above the threshold and their extraction coordinates
+
+        Parameters
+        ----------
+        slide : Slide
+            The slide to extract the tiles from.
+        extraction_mask : BinaryMask, optional
+            BinaryMask object defining how to compute a binary mask from a Slide.
+            Default `BiggestTissueBoxMask`.
+
+        Returns
+        -------
+        Tuple[List[Tuple[float, CoordinatePair]], List[Tuple[float, CoordinatePair]]]
+            List of tuples containing the scores and the extraction coordinates
+            for the filtered tiles. If scaled=True, each score `s_i` of
+            the i-th tile is normalized as
+
+            .. math::
+
+                s_{\hat{i}}=\frac{s_i-\min_{j\in T}{s_j}}{\max_{j\in T}{s_j}-\min_{j\in T}{s_j}}
+
+            where `T` is the set of all the retrieved tiles. Notice that the normalized
+            scores range between 0 and 1. This could be useful to have a more intuitive
+            comparison between the scores. Each tuple represents a tile.
+
+        Raises
+        ------
+        ValueError
+            If ``threshold`` is negative.
+        """  # noqa
+        all_scores = self._scores(slide, extraction_mask)
+        scaled_scores = self._scale_scores(all_scores)
+
+        if self.threshold < 0:
+            raise ValueError(f"'threshold' cannot be negative ({self.threshold})")
+
+        filtered_tiles_scores = []
+        filtered_tiles_scaled_scores = []
+
+        for i in range(len(all_scores)):
+            if all_scores[i][0] > self.threshold:
+                filtered_tiles_scores.append(all_scores[i])
+                filtered_tiles_scaled_scores.append(scaled_scores[i])
+
+        return filtered_tiles_scores, filtered_tiles_scaled_scores
+
+    @staticmethod
+    def _save_report(
+        report_path: str,
+        filtered_tiles_scores: List[Tuple[float, CoordinatePair]],
+        filtered_tiles_scaled_scores: List[Tuple[float, CoordinatePair]],
+        filenames: List[str],
+    ) -> None:
+        """Save to ``filename`` the report of the saved tiles with the associated score.
+
+        The CSV file
```

Comment:
oh, I see this line is cut (same as in the `ScoreTiler`). I'm not sure what we wanted to write here (?)
Could you remove this from both here and the `ScoreTiler`?